Minio/ApiEndpoints/ObjectOperations.cs

```diff
@@ -248,11 +248,17 @@ public partial class MinioClient : IObjectOperations
 
             double expectedReadSize = partSize;
             int partNumber;
+            int numPartsUploaded = 0;
             bool skipUpload = false;
             for (partNumber = 1; partNumber <= partCount; partNumber++)
             {
                 byte[] dataToCopy = ReadFull(data, (int)partSize);
-
+                if (dataToCopy == null)
+                {
+                    skipUpload = true;
```

Comment:
Looks like skipUpload is not read outside the for loop, so this line can be removed.

Reply:
good catch, removed that line.